doc/dune-files.rst

```diff
@@ -136,10 +136,9 @@ to use the :ref:`include_subdirs` stanza.
   advanced use only, by default you shouldn't set this option
 
 - ``(kind <kind>)`` is the kind of the library. The default is ``normal``, other
-  available choices are ``ppx_rewriter`` and ``ppx_deriver`` and must be set
-  when the library is intended to be used as a ppx rewriter or a ``[@@deriving
-  ...]`` plugin. The reason why ``ppx_rewriter`` and ``ppx_deriver`` are split
-  is historical and hopefully we won't need two options soon
+  available choices are ``ppx_deriver`` and ``ppx_rewriter``. ``ppx_deriver``
+  is used for ppx preprocessors using the ``ppx_deriving`` package and
```

Comment:
It's useful to explicitly name `ppx_deriving`, but I think that this should include something like "(or other packages handling [@@deriving])".

Reply:
I am still confused. I build a rewriter handling `[@@deriving something]` using `ppxlib`. Setting `(kind ppx_deriver)` causes problems. Using `(kind ppx_rewriter)` works fine. See [initial post](https://github.com/ocaml/dune/issues/1327#issue-363644006).

Reply:
Ah, sorry I missed that part. I am not sure what is the exact distinction then. @diml should be able to explain this better.

Reply:
I just added a comment in the other ticket